drivers/hmis/app/graphql/types/hmis_schema/custom_data_element.rb

```diff
@@ -22,7 +22,15 @@ class HmisSchema::CustomDataElement < Types::BaseObject
 
     def all_values(parent:)
       parent = load_ar_association(parent, :owner) if parent.is_a? Hmis::Hud::HmisService # special case for view
-      load_ar_association(object, :values, scope: parent.custom_data_elements)
+      ids = load_ar_association(parent, :custom_data_elements).pluck(:id)
+      # get all values existing for this CustomDataElementDefinition, and filter
+      # it down to only elemnts that are relevant to this parent
+      load_ar_association(object, :values).where(id: ids)
```

Comment:
So `load_ar_association` returns an AR scope that is preloaded. However if you call `where` or `pluck` on it (or some other method that maps to a DB operation), then the preloaded value is ignored and a new DB query is made. This is true for any preloaded scope in rails, not just the graphql preloader.

So I'd suggest either not preloading here (since we would discard the value anyway) or using array operations instead:
```suggestion
      ids = load_ar_association(parent, :custom_data_elements).map(&:id).to_set
      # get all values existing for this CustomDataElementDefinition, and filter
      # it down to only elemnts that are relevant to this parent
      load_ar_association(object, :values).filter { |v| v.id.in?(ids) }
```

Reply:
Ahh okay, I see. Super helpful explanation. I'll switch to array operations.